### PR TITLE
Enhancing test view for #25, fix for #29 test names with double quotes, setting test results on load

### PR DESCRIPTION
**Pull request by rythos42**
#22 , I found 2 issues:

1. I needed to have the extension call "yarn test", so that was an easy configuration.
2. The reported results needed to have their "test" value be equal to the id of the test being run. 

Previously it looks like "test" was set to be an object of some kind. I'm not sure how this was working for others? I'm using Test Explorer 2.3.2, maybe the existing code was working for a previous version? Maybe some more analysis needs to go into why no one else has reported this issue yet. I just wanted to run my tests. :) (and now they are)

#25 I wanted to be able to view the file list easier.

#26 While startup is really slow, we can mitigate this a bit by using the results from load and displaying them. In my usual workflow, I don't tend to run tests right away on startup anyway, so having it run the tests in the background for a bit and displaying the results seems like a decent idea while we look at speeding up load time.

I also 
- added a yarn.lock file, as it was added by yarn install.
- ran Prettier/ESLint on all files to improve readability

### CODE REVIEW

**Comment by iMarv** [comment]
Why not add prettier and tslint to the devDependencies to ensure that the style always matches a proper format?

**Comment by rythos42** [comment]
> Why not add prettier and tslint to the devDependencies to ensure that the style always matches a proper format?

I'm working on this on my company time, so am working on making it work for me first, and will address "quality of (dev) life" issues later. But first, I have to talk with the maintainer of this repo!

**Comment by rtbenfield** [comment]
Sorry for the late response, things have been busy on my end. I'm taking a look through your issue reports and changes now.

In regards to yarn.lock, Prettier, and ESLint:
* This project does not use Yarn, and instead relies on NPM's package-lock.json. I think it is best to stick with NPM for now to lower the barrier for anyone looking to contribute to the project.
* Instead of Prettier/ESLint, this project is using TSLint. I'd prefer to stick with TSLint for now, but I am open to starting a discussion about switching to ESLint since their TypeScript support has improved since the last time I used it.

**Comment by iMarv** [comment]
tslint is fine @rtbenfield, but would still consider prettier to ensure proper formatting all over the project

**Comment by rythos42** [comment]
I'll update my PR early this week to:

- Remove yarn.lock and update the package-lock.json
- Fix my merge conflict
- Switch to TSLint. I didn't know it was in the project, so added ESLint assuming it wasn't being done yet!

I'm with iMarv on Prettier - I like consistent formatting. But you're the project boss, so let me know if you want me to remove it too, I'm easy!

I'd also really like to hear if you looked at the commit for #22 -- I don't want to commit something that breaks everyone else.

**Comment by rythos42** [comment]
@rtbenfield I updated my PR as I said above.

File: src/adapter.ts [hunk 1]
```diff
@@ -97,42 +106,7 @@ export default class JestTestAdapter implements TestAdapter {
...
-          state: "running",
-          suite: mapJestFileResultToTestSuiteInfo(fileResult, this.workspace.uri.fsPath),
-          type: "suite",
-        } as TestSuiteEvent);
-
-        fileResult.assertionResults.forEach((assertionResult) => {
-          this.testStatesEmitter.fire({
-            decorations: mapJestAssertionToTestDecorations(assertionResult, fileResult.name, reconciler),
-            state: assertionResult.status,
-            test: mapJestAssertionToTestInfo(assertionResult, fileResult.name),
```
Comment (rythos42) [inline src/adapter.ts]: @rtbenfield I put this line back in in my local and it works. I suspect user error...I'll update the PR.

File: src/JestManager.ts [hunk 2]
```diff
@@ -70,23 +72,27 @@ export default class JestManager {
...
-      testNamePattern: testFilter && testFilter.testNamePattern
-        ? `"${testFilter.testNamePattern}"`
-        : undefined,
+      testFileNamePattern:
+        testFilter && testFilter.testFileNamePattern
+          ? `"${testFilter.testFileNamePattern}"`
+          : undefined,
+      testNamePattern:
+        testFilter && testFilter.testNamePattern
+          ? `"${testFilter.testNamePattern.replace(/"/g, '\\"')}"`
```
Comment (rythos42) [inline src/JestManager.ts]: This fixes #29

**Comment by rythos42** [comment]
@rtbenfield Are you interested in this PR? I'm interested in working on this project more, but I don't want to add more to this PR and yet if it's going into the project I need to build on top of it.

**Comment by rythos42** [comment]
Great, thank you @rtbenfield! Next time I get some time I want to work on taking your load performance changes and integrating them into the tree view.

**Comment by rtbenfield** [comment]
@rythos42  Sorry it took me so long to get time to review this. It looks great and I have merged the changes. I really appreciate you taking the time to contribute to this project. I think I am going to publish this as 0.2 since there is a behavioral change in the display of the tests. I'll just shift the remaining 0.2 tasks to 0.3.

I think we will also need to come up with some styling guidelines pretty soon. I noticed a lot of the line changes were from formatting and caused a lot of "noise". If you want to open a PR to add Prettier I would be interested in checking that out.

This may be a personal preference, but I like to try to make PRs specific to one issue unless it addresses multiple related issues. It makes it very easy to approve one change while continuing discussion on another and keeps the Git history clean. It would be great if we could make that a practice going forward. If you don't mind, I would like to tag you as a reviewer in any changes I make as well. I get nervous publishing code that only I have looked at 😄.

**Comment by rythos42** [comment]
I'll add an issue for Prettier and see about a PR for it.

On "a PR per issue", to be honest I'm not sure how to do that on GitHub. Would I make a branch in my fork and then make a PR from that branch? Then you would approve the PR and and you'd have my branch in your repo and merge it? Or would I make a PR from my branch into your master branch and then you'd approve it and then it's in master?

The latter is how we do BitBucket PRs, I guess it should work the same?

I'd be happy to be a reviewer!

**Comment by rtbenfield** [comment]
I'm new to this too, so I guess we will learn together. I think you have it right with the latter option. You can make a branch in your fork and when you open a PR you can select the branch rather than your master branch. When the PR gets approved, it would squash merge the changes from your branch into master branch of this codebase. Then you can update your fork's master with the latest from this one.

**Pull request merged**